libs/core---vm/scheduler.cpp

```diff
@@ -97,8 +97,12 @@ static void panic_core(int error_code) {
 extern "C" void target_panic(int error_code) {
     panic_core(error_code);
 
+#if defined(PXT_VM)
```

Comment:
The `while (1)` loop was locking pxt.dll, and would freeze Unity the next time it made a dll call. I'm not certain this is the best solution, but does improve things.

Reply:
I think that's cool. My comment was about allocating 1M always :)

Reply:
For pxt.dll/.so/.dylib, could it just use the regular heap? No strict size limitation?

Reply:
It could be it will be difficult to track which pointers are in heap (isReadOnly() intention). I would rather allocate 16M if needed. The memory situation is quite bi-modal - either you have ~100k or you have ~100M it seems.